model/riscv_sys_control.sail

```diff
@@ -340,8 +340,7 @@ function trap_handler(del_priv : Privilege, intr : bool, c : exc_code, pc : xlen
 
```

Comment:
There's a load of implicit CSR writes here that need to be tracked. We do it like this:

```
function track_trap(p : Privilege, intr : bool, c : exc_code) -> unit = {
  track_writeCSR(csr_name_map("mstatus"));
  match (p) {
    Machine => {
      track_writeCSR(csr_name_map("mcause"));
      track_writeCSR(csr_name_map("mtval"));
      track_writeCSR(csr_name_map("mepc"));
    },
    Supervisor => {
      track_writeCSR(csr_name_map("scause"));
      track_writeCSR(csr_name_map("stval"));
      track_writeCSR(csr_name_map("sepc"));
    },
    User => {
      track_writeCSR(csr_name_map("ucause"));
      track_writeCSR(csr_name_map("utval"));
      track_writeCSR(csr_name_map("uepc"));
    },
  };
}
```